satellite/satellitedb/usercredits.go

```diff
@@ -70,37 +68,58 @@ func (c *usercredits) Create(ctx context.Context, userCredit console.UserCredit)
 		referrerID = userCredit.ReferredBy[:]
 	}
 
-	var result sql.Result
-	var err error
+	var shouldCreate bool
+	switch userCredit.OfferInfo.Type {
+	case rewards.Partner:
+		shouldCreate = false
+	default:
+		shouldCreate = userCredit.OfferInfo.Status.IsDefault()
+	}
+
+	var dbExec interface {
+		ExecContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error)
+	}
+
+	if c.tx != nil {
```

Comment:
What will happen with other `usercredit` db calls, if there is a transaction in progress?

Reply:
We will use the existing transaction which is what this if statement used for

Reply:
I am talking about other methods like here:
https://github.com/storj/storj/blob/d33c4102567ed451beabcf21307575ab4cc9b86a/satellite/satellitedb/usercredits.go#L29

Reply:
or here: 
https://github.com/storj/storj/blob/d33c4102567ed451beabcf21307575ab4cc9b86a/satellite/satellitedb/usercredits.go#L177

Reply:
I can change it to also check if there's existing transaction

Reply:
Sure, just wanted to make sure that we are aware of the issue if it exists